core/backend/src/test/native/ECSchemaOps.test.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import * as path from "path";
+import { assert } from "chai";
+import { generateSchemaSha1Hash } from "../../ECSchemaOps";
+import { KnownTestLocations } from "../KnownTestLocations";
+
+describe("ECSchemaOps", () => {
+  let assetsDir: string;
+
+  before(async () => {
+    assetsDir = path.join(KnownTestLocations.assetsDir, "ECSchemaOps");
+  });
+
+  after(async () => {
+  });
+
+  it("computeChecksum", async () => {
+    const schemaPath = path.join(assetsDir, "SchemaA.ecschema.xml");
+    const refPath = path.dirname(schemaPath);
+    const sha1 = generateSchemaSha1Hash(schemaPath, [refPath]);
+    assert.isDefined(sha1);
+    assert.equal(sha1, "3ac6578060902aa0b8426b61d62045fdf7fa0b2b", "Expected sha1 hash values to match");
+  });
+
+  it("computeChecksumWithExactRefMatch", async () => {
+    const schemaPath = path.join(assetsDir, "SchemaA.ecschema.xml");
+    let refPath = path.dirname(schemaPath);
+    refPath = path.join(refPath, "exact-match");
+    const sha1 = generateSchemaSha1Hash(schemaPath, [refPath], true);
```

Comment:
probably a good idea to add a test that generates and exception too.

Reply:
> > We may call it outside the package it is defined in but we do not intend it to be called by 3rd parties and do not provide any guarantee API consistency between versions.
> 
> Given this, i strongly suggest this be an (undocumented) @internal method on `IModelHost`:
> 
> ```ts
> @internal
> public static computeSchemaChecksum(arg: { schemaXmlPath: string, referencePaths: string[], exactMatch?: boolean }) {
>   return this.platform.computeSchemaChecksum(arg);
> }
> ```

Sounds good.

Reply:
> probably a good idea to add a test that generates an exception too.

Yes, that's a good suggestion. I will also do that.